### PR TITLE
common,tools,crush,test: misc converity & klocwork fixes

### PR DESCRIPTION
**Pull request by Songweibin**
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

File: src/crush/mapper.c [hunk 1]
```diff
@@ -792,7 +792,7 @@ static void crush_choose_indep(const struct crush_map *map,
 							out2, rep,
 							recurse_tries, 0,
 							0, NULL, r, choose_args);
-						if (out2[rep] == CRUSH_ITEM_NONE) {
+						if (out2 && out2[rep] == CRUSH_ITEM_NONE) {
```
Comment (tchaikov) [inline src/crush/mapper.c]: shall we check for null at line 801, see https://github.com/ceph/ceph/pull/29316/commits/f575e060bcf2ae2ee414035c2435f54b5915ce6a#diff-0df13ad294f6585c322588cfe026d701R801

File: src/mount/mount.ceph.c [hunk 2]
```diff
@@ -40,6 +40,8 @@ static char *mount_resolve_src(const char *orig_str)
 	char *mount_path;
 	char *src;
 	char *buf = strdup(orig_str);
+	if (!buf)
```
Comment (tchaikov) [inline src/mount/mount.ceph.c]: i don't think the caller is expecting an errno. probably we can just print an error message and return `NULL` , like
```c++
fprintf(stderr, "%s: failed to allocate memory\n", __func__);
return NULL;
```

**Comment by Songweibin** [comment]
@tchaikov  Updated, please take a look, thank you.

**Comment by liewegas** [comment]
http://pulpito.ceph.com/sage-2019-07-26_20:58:43-rados-wip-sage-testing-2019-07-26-1403-distro-basic-smithi/

**Pull request merged**